highfidelity/hifi
# Include a marketplace status bar and access to "marketplace" of marketplaces

Marketplace now includes a status bar with access to a "marketplace" of marketplaces (a Hifi HTML page that links to Hifi's marketplace and Clara.io). The download process for Clara is the same as before, but now the status bar alerts the user when they attempt to download a model in a format that is not Autodesk FBX.

##  **Testing**

1. Load "clara.js" from system/marketplaces/clara.js
2. Open Marketplace. There may be two marketplace windows; disregard the one that points directly to Hifi Marketplace (this bug has been filed already). The other marketplace window should display an HTML page with two tiles: a Marketplace tile and a Clara.io tile. 
3. The bottom status bar should be a light blue color, with an info icon and the text "Check out other marketplaces." There should also be a gray "SEE ALL MARKETS" right-anchored on the status bar.
4. Click the "Explore" button for Hifi marketplace; you should be redirected to hifi's marketplace and be able to download a model from there.
5. Click on the "SEE ALL MARKETS" button located on the bottom right of the Marketplace overlay. You should be redirected back to the "marketplaces" html page.
6. Now click on the "explore" button for Clara.io. You should be redirected to Clara.io's library page, where the "Display only downloadable and editable models" option is checked off. Also, the status bar should now read "Choose a model and click Download -> Autodesk FBX."
7. Choose any model you like, or search for a model in their given search bar.
8. A new page should open up with your model. Go to the "Download" button, and click.
9. You will see a list with several options for file types. Skip the Autodesk FBX option, and select the next option.
10. An overlay should appear that exports the model, then a yellow "Download" button should appear. If you hover over the button, the overlay should disappear and the status bar should turn red, display an alert icon, and read "High Fidelity only supports Autodesk FBX models."
11. Wait 9 seconds; the status bar should revert back to a light blue, an info icon, and the text "Choose a model and click Download -> Autodesk FBX."
12. Repeat steps 8-11 for all of the other model types except for FBX; the results should be the same.
13. Finally, click on Download and click FBX. This time, click on the yellow "Download" button in the overlay. A new window should appear briefly, the exporting overlay you just clicked on should close, then the Asset browser will pop up directing you to save your model. From here, you can rename your model then add it to world if you like (all of this has been tested thoroughly, so this is just a sanity check), or you can just press cancel and return to the marketplace.
14. Click on the "SEE ALL MARKETS" button in the status bar. You should return to the marketplaces HTML page, and the status should read "Check out other marketplaces." 
15. You're done!
